src/renderer/hooks/useTestImport.ts

```diff
@@ -5,10 +5,14 @@ import { map } from "lodash";
 import { v4 as uuidv4 } from "uuid";
 import { ImportTestSettings } from "@/renderer/routes/test_sequencer_panel/components/ImportTestModal";
 
-function parseDiscoverContainer(data: TestDiscoverContainer) {
+function parseDiscoverContainer(
+  data: TestDiscoverContainer,
+  sourceCode: string,
+) {
   return map(data.response, (container) => {
     const new_elem: Test = {
       ...container,
+      sourceCode,
```

Comment:
how will we handle this when the user wants to change the python code in the file? 

Suggestions:
- use the reactive paradigm: detect or continuously monitor for changes in the file and define a callback function to populate the "sourceCode" field for an element.

Reply:
@izi-on Very good question. I'd go with fetching code content each time modal is open as this seems to be simplest yet effective..